base/iddict.jl

```diff
@@ -132,8 +132,8 @@ function delete!(d::IdDict{K}, @nospecialize(key)) where K
     d
 end
 
-function empty!(d::IdDict)
-    resize!(d.ht, 32)
+function empty!(d::IdDict; preserve_size=false)
```

Comment:
In my opinion, this should be in a separate PR while all the non API performance improvement can stay here.